src/hid_linux.c

```diff
@@ -268,11 +271,24 @@ fido_hid_open(const char *path)
 			fido_hid_close(ctx);
 			return (NULL);
 		}
-		interval_ms = retries * 100000000L;
-		tv_pause.tv_sec = interval_ms / 1000000000L;
-		tv_pause.tv_nsec = interval_ms % 1000000000L;
-		if (nanosleep(&tv_pause, NULL) == -1) {
-			fido_log_error(errno, "%s: nanosleep", __func__);
+		if (clock_gettime(CLOCK_MONOTONIC, &tv_pause) == -1) {
```

Comment:
Maybe we can add a comment here like:
```
// Take the current time and add the required timeout so that we can use it with `TIMER_ABSTIME`
// below. We can't use nanosleep with a relative timer, because we might be interrupted multiple
// times and would then have to adjust for the time difference after the interrupt.
```

Not sure of the exact wording, but at least I was wondering at first why we now need to use `clock_gettime`.

Reply:
Good idea. Done,